### PR TITLE
Cleanup axes_rgb.

### PR DESCRIPTION
**Pull request by anntzer**
- `imshow_rgb` is just `imshow` after stacking the r, g, and b channels
  (the docstring of `np.dstack` even explicitly mentions this use).
- RGBAxesBase only differs from RGBAxes by not having a _defaultAxesClass,
  and thus requiring specific handling for catching that case in the
  constructor; simplify this by getting rid of the base class (other
  subclasses of RGBAxes can still override _defaultAxesClass as needed).
- The constructor of `RGBAxes` is duplicated from make_axes_rgb; just
  call the latter instead; this makes add_RGB_to_figure unused so
  deprecate it as well (it's clearly just an internal helper).

## PR Checklist

- [ ] Has Pytest style unit tests
- [ ] Code is [Flake 8](http://flake8.pycqa.org/en/latest/) compliant
- [ ] New features are documented, with examples if plot related
- [ ] Documentation is sphinx and numpydoc compliant
- [ ] Added an entry to doc/users/next_whats_new/ if major new feature (follow instructions in README.rst there)
- [ ] Documented in doc/api/api_changes.rst if API changed in a backward-incompatible way

### CODE REVIEW

File: lib/mpl_toolkits/axisartist/axes_rgb.py [hunk 1]
```diff
@@ -1,8 +1,6 @@
-from mpl_toolkits.axes_grid1.axes_rgb import (
-    make_rgb_axes, imshow_rgb, RGBAxesBase)
-
+from mpl_toolkits.axes_grid1.axes_rgb import make_rgb_axes, imshow_rgb, RGBAxes
 from .axislines import Axes
 
 
-class RGBAxes(RGBAxesBase):
+class RGBAxes(RGBAxes):
```
Comment (timhoffm) [inline lib/mpl_toolkits/axisartist/axes_rgb.py]: ??? auto-relplaced?

File: lib/mpl_toolkits/axes_grid/axes_rgb.py [hunk 2]
```diff
@@ -1,7 +1,6 @@
-from mpl_toolkits.axes_grid1.axes_rgb import (
-    make_rgb_axes, imshow_rgb, RGBAxesBase)
+from mpl_toolkits.axes_grid1.axes_rgb import make_rgb_axes, imshow_rgb, RGBAxes
 from mpl_toolkits.axisartist.axislines import Axes
 
 
-class RGBAxes(RGBAxesBase):
+class RGBAxes(RGBAxes):
```
Comment (timhoffm) [inline lib/mpl_toolkits/axes_grid/axes_rgb.py]: ??? auto-replaced?

File: lib/mpl_toolkits/axisartist/axes_rgb.py (hunk 1 above)
Comment (anntzer) [reply lib/mpl_toolkits/axisartist/axes_rgb.py]: it's inheriting from the RGBAxes class imported from the other (base) module, which happens *just* above.

File: lib/mpl_toolkits/axes_grid/axes_rgb.py (hunk 2 above)
Comment (anntzer) [reply lib/mpl_toolkits/axes_grid/axes_rgb.py]: as above

Comment (timhoffm) [reply lib/mpl_toolkits/axes_grid/axes_rgb.py]: Oh, I see. I would rather use a different name to make it less confuing. Either
~~~
from mpl_toolkits.axes_grid1.axes_rgb import RGBAxes as _RGBAxes
class RGBAxes(_RGBAxes):
~~~
or 
~~~
import mpl_toolkits.axes_grid1.axes_rgb
class RGBAxes(mpl_toolkits.axes_grid1.axes_rgb.RGBAxes):
~~~

**Comment by anntzer** [comment]
> Didn't quite follow if the tick label handling in the constructor of RGBAxes and in make_rgb_axes is really equivalent. The code is different, but I suppose the effect is the same. I rely on that you have checked this.

One does
```
        for t in ax1.yaxis.get_ticklabels() + ax1.xaxis.get_ticklabels():
            t.set_visible(False)
        try:
            for axis in ax1.axis.values():
                axis.major_ticklabels.set_visible(False)
        except AttributeError:
            pass
```
the other does (yuck, mpl_toolkits-specific API)
```
            ax1.axis[:].toggle(ticklabels=False)
```
which basically calls `set_tick_params(label1/2On=False)` so I *think* things are the same.

File: lib/mpl_toolkits/axes_grid/axes_rgb.py (hunk 2 above)
Comment (anntzer) [reply lib/mpl_toolkits/axes_grid/axes_rgb.py]: Sure.  I changed axisartist as you suggested and made axes_grid just star-import from axisartist, as 1) they're exactly the same, 2) axes_grid (not axes_grid1) is deprecated anyways, and 3) star-importing from axisartist is likewise used for other parts of axes_grid, e.g. angle_helper.

File: lib/mpl_toolkits/axes_grid1/axes_rgb.py [hunk 3]
```diff
@@ -55,6 +56,7 @@ def make_rgb_axes(ax, pad=0.01, axes_class=None, add_all=True):
     return ax_rgb
 
 
+@cbook.deprecated("3.3", alternative="ax.imshow(np.dstack([r, g, b]))")
```
Comment (QuLogic) [inline lib/mpl_toolkits/axes_grid1/axes_rgb.py]: Can we re-implement the function that way, too? It's currently pretty inefficient.

File: lib/mpl_toolkits/axes_grid1/axes_rgb.py [hunk 4]
```diff
@@ -112,47 +117,12 @@ def __init__(self, *args, pad=0, add_all=True, **kwargs):
...
-        divider.set_horizontal([Size.AxesX(ax), pad_size, xsize])
-        divider.set_vertical([ysize, pad_size, ysize, pad_size, ysize])
-
-        ax.set_axes_locator(divider.new_locator(0, 0, ny1=-1))
-
-        ax_rgb = []
-        for ny in [4, 2, 0]:
-            ax1 = axes_class(ax.get_figure(),
-                             ax.get_position(original=True),
-                             sharex=ax, sharey=ax, **kwargs)
```
Comment (QuLogic) [inline lib/mpl_toolkits/axes_grid1/axes_rgb.py]: This `kwargs` is missing from `make_rgb_axes`.

File: lib/mpl_toolkits/axes_grid1/axes_rgb.py (hunk 3 above)
Comment (anntzer) [reply lib/mpl_toolkits/axes_grid1/axes_rgb.py]: done

File: lib/mpl_toolkits/axes_grid1/axes_rgb.py (hunk 4 above)
Comment (anntzer) [reply lib/mpl_toolkits/axes_grid1/axes_rgb.py]: good catch, fixed.

**Pull request merged**